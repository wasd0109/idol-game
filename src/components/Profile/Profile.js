import React from 'react';

import profileImage from '../../assets/profile_image.jpg';

const name = 'Pekorin';
const element = 'Fire';
const attack = 23;
const defense = 30;
const magicAttack = 30;
const magicDefense = 40;
const agility = 10;
const luck = 5;

const stats = [
  ['name', name],
  ['element', element],
  ['attack', attack],
  ['defense', defense],
  ['magic attack', magicAttack],
  ['magic defense', magicDefense],
  ['agility', agility],
  ['luck', luck],
];

const actions = ['Practice', 'Live'];

function Profile() {
  return (
    <div>
      <div className="mx-4 my-2" id="stats">
        <ul className="flex flex-wrap w-4/12 border-l-2 border-t-2">
          <img
            src={profileImage}
            alt=""
            className="w-12/12 border-l-0 border-r-2 border-b-2 m-auto"
          />
          {stats.map((stat) => (
            <li className="w-6/12 capitalize border-r-2 border-b-2">
              {stat[0]} <span className="text-lg font-bold">{stat[1]}</span>
            </li>
          ))}
        </ul>
      </div>
      <div></div>
    </div>
  );
}

export default Profile;
