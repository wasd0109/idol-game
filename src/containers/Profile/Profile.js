import React from 'react';
import Stats from '../../components/Stats';
import Actions from '../../components/Actions';

function Profile({ char }) {
  const stats = [];
  for (const [key, value] of Object.entries(char)) {
    stats.push([key.replace('_', ' '), value]);
  }

  return (
    <div className="flex mx-4 my-2 flex-wrap">
      <Stats stats={stats} />
      <Actions />
    </div>
  );
}

export default Profile;
