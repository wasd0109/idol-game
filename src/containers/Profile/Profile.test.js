import React from 'react';
import { render, screen, waitFor, waitForElement, } from '@testing-library/react';
import fetchMock from 'jest-fetch-mock'
import Profile from './Profile';
import { act } from 'react-dom/test-utils';
import userEvent from '@testing-library/user-event';

const initialProps = {
    userID: "0",
}

let container;

beforeEach(async () => await act(async () => {
    container = render(<Profile {...initialProps} />)
}))


describe("Profile container render correctly", () => {
    test("Profile container render", async () => {
        expect(container).toMatchSnapshot();
    })
})

describe("Profile container function correctly", () => {
    // test("Profile container fetch player stats correctly", () => {
    //     const check = jest.fn();
    //     fetchMock.mockResponseOnce(req => {
    //         if (req.url === `https://idol-game.herokuapp.com/profile/${initialProps.userID}`) {
    //             check()
    //         }
    //         return Promise.resolve("Success")
    //     });
    //     waitFor(() => { console.log(check.mock.calls.length); expect(check).toBeCalled() })
    // })

    // test("Passing action result to Actions component", async () => {
    //     fetchMock.mockResponseOnce(() => Promise.resolve(JSON.stringify("Success")));
    //     const button = screen.getByText("Practice")
    //     await act(async () => { userEvent.click(button) });
    //     expect(screen.getByText("Success")).toBeInTheDocument();
    // })

    // test("Performing action trigger stats reload", async () => {
    //     const check = jest.fn();
    //     fetchMock.mockResponseOnce(req => {
    //         if (req.url === `https://idol-game.herokuapp.com/profile/${initialProps.userID}`) {
    //             check();
    //         }
    //         return Promise.resolve("Success")
    //     })
    //     const button = screen.getByText("Practice")
    //     await act(async () => { userEvent.click(button) });
    //     waitFor(() => expect(check).toBeCalled())
    // })
})