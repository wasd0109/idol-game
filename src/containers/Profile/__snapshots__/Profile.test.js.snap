// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile container render correctly Profile container render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex mx-4 my-2 flex-wrap"
      >
        <div
          class="md:w-4/12 xl:w-3/12 h-auto"
          id="stats"
        >
          <ul
            class="flex flex-wrap border-l-2 border-t-2"
          >
            <div
              class="w-12/12 h-auto border-l-0 border-r-2 border-b-2 m-auto"
            >
              <img
                alt="Profile"
                src="profile_image.jpg"
              />
            </div>
            <li
              class="w-6/12 capitalize border-r-2 border-b-2"
            >
              name
               
              <span
                class="text-lg font-bold"
                id="skeleton"
              >
                <span>
                  <span
                    class="react-loading-skeleton css-1vmnjpn-skeletonStyles-Skeleton"
                    style="width: 40px;"
                  >
                    ‌
                  </span>
                </span>
              </span>
            </li>
            <li
              class="w-6/12 capitalize border-r-2 border-b-2"
            >
              title
               
              <span
                class="text-lg font-bold"
                id="skeleton"
              >
                <span>
                  <span
                    class="react-loading-skeleton css-1vmnjpn-skeletonStyles-Skeleton"
                    style="width: 40px;"
                  >
                    ‌
                  </span>
                </span>
              </span>
            </li>
            <li
              class="w-6/12 capitalize border-r-2 border-b-2"
            >
              HP
               
              <span
                class="text-lg font-bold"
                id="skeleton"
              >
                <span>
                  <span
                    class="react-loading-skeleton css-1vmnjpn-skeletonStyles-Skeleton"
                    style="width: 40px;"
                  >
                    ‌
                  </span>
                </span>
              </span>
            </li>
            <li
              class="w-6/12 capitalize border-r-2 border-b-2"
            >
              level
               
              <span
                class="text-lg font-bold"
                id="skeleton"
              >
                <span>
                  <span
                    class="react-loading-skeleton css-1vmnjpn-skeletonStyles-Skeleton"
                    style="width: 40px;"
                  >
                    ‌
                  </span>
                </span>
              </span>
            </li>
            <li
              class="w-6/12 capitalize border-r-2 border-b-2"
            >
              element
               
              <span
                class="text-lg font-bold"
                id="skeleton"
              >
                <span>
                  <span
                    class="react-loading-skeleton css-1vmnjpn-skeletonStyles-Skeleton"
                    style="width: 40px;"
                  >
                    ‌
                  </span>
                </span>
              </span>
            </li>
            <li
              class="w-6/12 capitalize border-r-2 border-b-2"
            >
              exp
               
              <span
                class="text-lg font-bold"
                id="skeleton"
              >
                <span>
                  <span
                    class="react-loading-skeleton css-1vmnjpn-skeletonStyles-Skeleton"
                    style="width: 40px;"
                  >
                    ‌
                  </span>
                </span>
              </span>
            </li>
            <li
              class="w-6/12 capitalize border-r-2 border-b-2"
            >
              attack
               
              <span
                class="text-lg font-bold"
                id="skeleton"
              >
                <span>
                  <span
                    class="react-loading-skeleton css-1vmnjpn-skeletonStyles-Skeleton"
                    style="width: 40px;"
                  >
                    ‌
                  </span>
                </span>
              </span>
            </li>
            <li
              class="w-6/12 capitalize border-r-2 border-b-2"
            >
              defense
               
              <span
                class="text-lg font-bold"
                id="skeleton"
              >
                <span>
                  <span
                    class="react-loading-skeleton css-1vmnjpn-skeletonStyles-Skeleton"
                    style="width: 40px;"
                  >
                    ‌
                  </span>
                </span>
              </span>
            </li>
            <li
              class="w-6/12 capitalize border-r-2 border-b-2"
            >
              magic attack
               
              <span
                class="text-lg font-bold"
                id="skeleton"
              >
                <span>
                  <span
                    class="react-loading-skeleton css-1vmnjpn-skeletonStyles-Skeleton"
                    style="width: 40px;"
                  >
                    ‌
                  </span>
                </span>
              </span>
            </li>
            <li
              class="w-6/12 capitalize border-r-2 border-b-2"
            >
              magic defense
               
              <span
                class="text-lg font-bold"
                id="skeleton"
              >
                <span>
                  <span
                    class="react-loading-skeleton css-1vmnjpn-skeletonStyles-Skeleton"
                    style="width: 40px;"
                  >
                    ‌
                  </span>
                </span>
              </span>
            </li>
            <li
              class="w-6/12 capitalize border-r-2 border-b-2"
            >
              agility
               
              <span
                class="text-lg font-bold"
                id="skeleton"
              >
                <span>
                  <span
                    class="react-loading-skeleton css-1vmnjpn-skeletonStyles-Skeleton"
                    style="width: 40px;"
                  >
                    ‌
                  </span>
                </span>
              </span>
            </li>
            <li
              class="w-6/12 capitalize border-r-2 border-b-2"
            >
              luck
               
              <span
                class="text-lg font-bold"
                id="skeleton"
              >
                <span>
                  <span
                    class="react-loading-skeleton css-1vmnjpn-skeletonStyles-Skeleton"
                    style="width: 40px;"
                  >
                    ‌
                  </span>
                </span>
              </span>
            </li>
          </ul>
        </div>
        <div
          class="w-full my-2 md:my-0 md:w-7/12 m-auto"
        >
          <div
            class="border-2 p-2 overflow-y-scroll capitalize"
            id="performed"
          />
          <div
            class="flex justify-evenly my-2 flex-wrap md:no-wrap"
            id="actions"
          >
            <button
              class="action-button bg-blue-500 my-2 md:m-0 hover:bg-blue-400 p-2 rounded w-5/12 md:w-2/12"
              id="Practice"
            >
              <p
                class="text-white font-bold"
              >
                Practice
              </p>
            </button>
            <button
              class="action-button bg-blue-500 my-2 md:m-0 hover:bg-blue-400 p-2 rounded w-5/12 md:w-2/12"
              id="Live"
            >
              <p
                class="text-white font-bold"
              >
                Live
              </p>
            </button>
            <button
              class="action-button bg-blue-500 my-2 md:m-0 hover:bg-blue-400 p-2 rounded w-5/12 md:w-2/12"
              id="Tsunagari"
            >
              <p
                class="text-white font-bold"
              >
                Tsunagari
              </p>
            </button>
            <button
              class="action-button bg-blue-500 my-2 md:m-0 hover:bg-blue-400 p-2 rounded w-5/12 md:w-2/12"
              id="Tweet"
            >
              <p
                class="text-white font-bold"
              >
                Tweet
              </p>
            </button>
          </div>
          <div
            class="items-center bg-blue-500 text-white font-semibold px-4 py-3 rounded"
            role="alert"
          >
            Press button to perform action
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex mx-4 my-2 flex-wrap"
    >
      <div
        class="md:w-4/12 xl:w-3/12 h-auto"
        id="stats"
      >
        <ul
          class="flex flex-wrap border-l-2 border-t-2"
        >
          <div
            class="w-12/12 h-auto border-l-0 border-r-2 border-b-2 m-auto"
          >
            <img
              alt="Profile"
              src="profile_image.jpg"
            />
          </div>
          <li
            class="w-6/12 capitalize border-r-2 border-b-2"
          >
            name
             
            <span
              class="text-lg font-bold"
              id="skeleton"
            >
              <span>
                <span
                  class="react-loading-skeleton css-1vmnjpn-skeletonStyles-Skeleton"
                  style="width: 40px;"
                >
                  ‌
                </span>
              </span>
            </span>
          </li>
          <li
            class="w-6/12 capitalize border-r-2 border-b-2"
          >
            title
             
            <span
              class="text-lg font-bold"
              id="skeleton"
            >
              <span>
                <span
                  class="react-loading-skeleton css-1vmnjpn-skeletonStyles-Skeleton"
                  style="width: 40px;"
                >
                  ‌
                </span>
              </span>
            </span>
          </li>
          <li
            class="w-6/12 capitalize border-r-2 border-b-2"
          >
            HP
             
            <span
              class="text-lg font-bold"
              id="skeleton"
            >
              <span>
                <span
                  class="react-loading-skeleton css-1vmnjpn-skeletonStyles-Skeleton"
                  style="width: 40px;"
                >
                  ‌
                </span>
              </span>
            </span>
          </li>
          <li
            class="w-6/12 capitalize border-r-2 border-b-2"
          >
            level
             
            <span
              class="text-lg font-bold"
              id="skeleton"
            >
              <span>
                <span
                  class="react-loading-skeleton css-1vmnjpn-skeletonStyles-Skeleton"
                  style="width: 40px;"
                >
                  ‌
                </span>
              </span>
            </span>
          </li>
          <li
            class="w-6/12 capitalize border-r-2 border-b-2"
          >
            element
             
            <span
              class="text-lg font-bold"
              id="skeleton"
            >
              <span>
                <span
                  class="react-loading-skeleton css-1vmnjpn-skeletonStyles-Skeleton"
                  style="width: 40px;"
                >
                  ‌
                </span>
              </span>
            </span>
          </li>
          <li
            class="w-6/12 capitalize border-r-2 border-b-2"
          >
            exp
             
            <span
              class="text-lg font-bold"
              id="skeleton"
            >
              <span>
                <span
                  class="react-loading-skeleton css-1vmnjpn-skeletonStyles-Skeleton"
                  style="width: 40px;"
                >
                  ‌
                </span>
              </span>
            </span>
          </li>
          <li
            class="w-6/12 capitalize border-r-2 border-b-2"
          >
            attack
             
            <span
              class="text-lg font-bold"
              id="skeleton"
            >
              <span>
                <span
                  class="react-loading-skeleton css-1vmnjpn-skeletonStyles-Skeleton"
                  style="width: 40px;"
                >
                  ‌
                </span>
              </span>
            </span>
          </li>
          <li
            class="w-6/12 capitalize border-r-2 border-b-2"
          >
            defense
             
            <span
              class="text-lg font-bold"
              id="skeleton"
            >
              <span>
                <span
                  class="react-loading-skeleton css-1vmnjpn-skeletonStyles-Skeleton"
                  style="width: 40px;"
                >
                  ‌
                </span>
              </span>
            </span>
          </li>
          <li
            class="w-6/12 capitalize border-r-2 border-b-2"
          >
            magic attack
             
            <span
              class="text-lg font-bold"
              id="skeleton"
            >
              <span>
                <span
                  class="react-loading-skeleton css-1vmnjpn-skeletonStyles-Skeleton"
                  style="width: 40px;"
                >
                  ‌
                </span>
              </span>
            </span>
          </li>
          <li
            class="w-6/12 capitalize border-r-2 border-b-2"
          >
            magic defense
             
            <span
              class="text-lg font-bold"
              id="skeleton"
            >
              <span>
                <span
                  class="react-loading-skeleton css-1vmnjpn-skeletonStyles-Skeleton"
                  style="width: 40px;"
                >
                  ‌
                </span>
              </span>
            </span>
          </li>
          <li
            class="w-6/12 capitalize border-r-2 border-b-2"
          >
            agility
             
            <span
              class="text-lg font-bold"
              id="skeleton"
            >
              <span>
                <span
                  class="react-loading-skeleton css-1vmnjpn-skeletonStyles-Skeleton"
                  style="width: 40px;"
                >
                  ‌
                </span>
              </span>
            </span>
          </li>
          <li
            class="w-6/12 capitalize border-r-2 border-b-2"
          >
            luck
             
            <span
              class="text-lg font-bold"
              id="skeleton"
            >
              <span>
                <span
                  class="react-loading-skeleton css-1vmnjpn-skeletonStyles-Skeleton"
                  style="width: 40px;"
                >
                  ‌
                </span>
              </span>
            </span>
          </li>
        </ul>
      </div>
      <div
        class="w-full my-2 md:my-0 md:w-7/12 m-auto"
      >
        <div
          class="border-2 p-2 overflow-y-scroll capitalize"
          id="performed"
        />
        <div
          class="flex justify-evenly my-2 flex-wrap md:no-wrap"
          id="actions"
        >
          <button
            class="action-button bg-blue-500 my-2 md:m-0 hover:bg-blue-400 p-2 rounded w-5/12 md:w-2/12"
            id="Practice"
          >
            <p
              class="text-white font-bold"
            >
              Practice
            </p>
          </button>
          <button
            class="action-button bg-blue-500 my-2 md:m-0 hover:bg-blue-400 p-2 rounded w-5/12 md:w-2/12"
            id="Live"
          >
            <p
              class="text-white font-bold"
            >
              Live
            </p>
          </button>
          <button
            class="action-button bg-blue-500 my-2 md:m-0 hover:bg-blue-400 p-2 rounded w-5/12 md:w-2/12"
            id="Tsunagari"
          >
            <p
              class="text-white font-bold"
            >
              Tsunagari
            </p>
          </button>
          <button
            class="action-button bg-blue-500 my-2 md:m-0 hover:bg-blue-400 p-2 rounded w-5/12 md:w-2/12"
            id="Tweet"
          >
            <p
              class="text-white font-bold"
            >
              Tweet
            </p>
          </button>
        </div>
        <div
          class="items-center bg-blue-500 text-white font-semibold px-4 py-3 rounded"
          role="alert"
        >
          Press button to perform action
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
